py-{astroid,importlib-metadata,pandas,pylint,qtconsole}: update to latest versions

reneeotten opened the pull request:
#### Description
Update various Python ports maintained by @stromnov to their latest versions: `py-astroid`, `py-importlib-metadata`, `py-pandas`, `py-pylint`, `py-qtconsole` (see commit messages).

###### Tested on
macOS 10.15.7 19H1217 x86_64
Xcode 12.4 12D4e

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

stromnov commented:
astroid 2.6.x:
> setuptools_scm`` has been removed and replaced by ``tbump`` in order to not have hidden runtime dependencies to setuptools

reneeotten commented:
> astroid 2.6.x:
> 
> > `setuptools_scm` has been removed and replaced by`tbump` in order to not have hidden runtime dependencies to setuptools

@stromnov true, I missed that... I removed the dependency on `py-setuptools_scm`, adding `py-tbump` is not needed.
